docs/extensibility.md

```diff
@@ -1,453 +1,275 @@
+utilizes exposed devices. This document will also cover the steps to get your Discovery Handler and broker added to
+Akri, should you wish to [contribute them back](./contributing.md).
+
+Before continuing, please read the [Akri architecture](./architecture.md), [Akri agent](./agent-in-depth.md), and
+[development](./development.md) documentation pages.  They will provide a good understanding of Akri, how it works, what
+components it is composed of, and how to build it.
+> **Note:** a Discovery Handler can use any set of steps to discover devices. It does not have to be a "protocol" in the
+> traditional sense. For example, Akri defines udev (not often called a "protocol") and OPC UA as protocols.
+
+Here, we will create a Discovery Handler to discover **HTTP-based devices** that publish random sensor data.
+
+Any Docker-compatible container registry will work for hosting the containers being used in this example (Docker Hub,
+Github Container Registry, Azure Container Registry, etc).  Here, we are using the [GitHub Container
+Registry](https://github.blog/2020-09-01-introducing-github-container-registry/). You can follow the [getting started
+guide here to enable it for
+yourself](https://docs.github.com/en/free-pro-team@latest/packages/getting-started-with-github-container-registry).
+
+> **Note:** if your container registry is private, you will need to create a kubernetes secret (`kubectl create secret
+> docker-registry crPullSecret --docker-server=<cr>  --docker-username=<cr-user> --docker-password=<cr-token>`) and
+> access it with an `imagePullSecret`.  Here, we will assume the secret is named `crPullSecret`.
 
 ## New DiscoveryHandler implementation
-If the resource you are interested in defining is not accessible through the [included protocols](./roadmap.md#currently-supported-protocols), then you will need to create a DiscoveryHandler for your new protocol.  Here, we will create a discovery handler in order to discover HTTP resources.
-
-New protocols require new implementations of the DiscoveryHandler:
-
+### Use `cargo generate` to clone the Discovery Handler template
+Pull down the [Discovery Handler template](https://github.com/kate-goldenring/akri-discovery-handler-template) using
+[`cargo-generate`](https://github.com/cargo-generate/cargo-generate). 
+```sh 
+cargo install cargo-generate
+cargo generate --git https://github.com/kate-goldenring/akri-discovery-handler-template.git --name akri-http-discovery-handler
+```
+### Specify the protocol name and locality of the devices it discovers
+Inside the newly created `akri-http-discovery-handler` project, navigate to `main.rs`. It contains all the logic to
+register our DiscoveryHandler with the Akri Agent. We only need to specify the protocol name and locality of our
+Discovery Handler. Set the protocol name to `http` and `shared` to true, as our HTTP Discovery Handler will discover
```

Comment:
Maybe show in code setting name and shared?  Or include the actual variable names to change.

Reply:
Added in variable names in 34e97dcf0